core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/evaluationsteps/LeftJoinQueryEvaluationStep.java

```diff
@@ -61,18 +55,60 @@ public CloseableIteration<BindingSet, QueryEvaluationException> evaluate(Binding
 		} else {
 			condition = null;
 		}
-		return new LeftJoinQueryEvaluationStep(right, condition, left, leftJoin, optionalVarCollector);
+		return new LeftJoinQueryEvaluationStep(right, condition, left, leftJoin, optionalVarCollector.getVarNames());
 	}
 
+	@Deprecated(forRemoval = true, since = "4.1.0")
 	public LeftJoinQueryEvaluationStep(QueryEvaluationStep right, QueryValueEvaluationStep condition,
-			QueryEvaluationStep left, LeftJoin leftJoin, VarNameCollector optionalVarCollector) {
+			QueryEvaluationStep left, LeftJoin leftJoin,
+			org.eclipse.rdf4j.query.algebra.helpers.VarNameCollector optionalVarCollector) {
```

Comment:
This is deprecated because we have replaced `org.eclipse.rdf4j.query.algebra.helpers.VarNameCollector`.